src/exporter/exporter.go

```diff
@@ -119,22 +118,59 @@ func (e *Exporter) createJobObject() error {
 		uint32(unsafe.Sizeof(jobInfo)),
 	)
 	if err != nil {
-		return fmt.Errorf("failed to set job object info:%v", err)
+		return fmt.Errorf("failed to set job object info: %v", err)
 	}
 
-	err = windows.AssignProcessToJobObject(e.jobObject, (*process)(unsafe.Pointer(e.cmd.Process)).handle)
+	handle, err := e.getHandle()
+	if err != nil {
+		return fmt.Errorf("failed to get proc handle: %v", err)
+	}
+
+	err = windows.AssignProcessToJobObject(e.jobObject, handle)
 	if err != nil {
 		return fmt.Errorf("failed to assign process to job object:%v", err)
 	}
 
 	return nil
 }
 
+func (e *Exporter) getHandle() (windows.Handle, error) {
+	var processHandle windows.Handle
+	if e.cmd.Process == nil {
+		return processHandle, fmt.Errorf("process cannot be nil pointer")
```

Comment:
Is `processHandle` an interface? It might be good to return a pointer so we can return `nil` when an error occurs, rather than a non-meaningful zero-value.

Reply:
is an uint